PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/TelemetryToggleAction.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.Presentation;
+import com.intellij.openapi.project.DumbAware;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.Project;
+import com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool.TelemetryClientProvider.TelemetryClientProviderVisitor;
+
+/**
+ * This class is responsible for toggling the telemetry service on and off.
+ */
+public class TelemetryToggleAction extends AnAction implements DumbAware {
+
+    // This is the telemetry service that will be toggled on and off.
+    private static TelemetryClientProviderVisitor telemetryService;
+
+    // This method sets the telemetry service
+    public static void setTelemetryService(TelemetryClientProviderVisitor service) {
+        telemetryService = service;
+    }
+
+    /**
+     * This method is called when the action is performed.
+     * It toggles the telemetry service on and off.
+     */
+    @Override
+    public void actionPerformed(AnActionEvent e) {
+        Project project = e.getProject();
+        if (project == null || telemetryService == null) {
+            return;
+        }
+
+        // If the telemetry service is running, stop it. Otherwise, start it.
+        if (telemetryService.isRunning()) {
+            telemetryService.stopTelemetryService();
+        } else {
+            telemetryService.startTelemetryService();
+        }
+        // Update the presentation of the action.
+        updatePresentation(e.getPresentation());
+    }
+
+    /**
+     * This method updates the presentation of the action.
```

Comment:
Not sure what this means, reword?

Reply:
ohh this relates to the UI representation -- to make sure it reflects the current state / get updated after an action

Reply:
added a more descriptive javadoc